py/server/deephaven/time.py

```diff
@@ -155,1885 +135,485 @@ def time_zone_alias_rm(alias: str) -> bool:
+        if dt is None:
             return None
-
-        if from_jtype(dt.getClass()) == ZonedDateTime:
-            dt = to_instant(dt)
-
-        return _JDateTimeUtils.toLocalTime(dt, tz)
+        elif isinstance(dt, LocalTime.j_type):
+            return datetime.time(dt.getHour(), dt.getMinute(), dt.getSecond(), dt.getNano() // 1000)
+        elif isinstance(dt, ZonedDateTime.j_type):
+            return datetime.time(dt.getHour(), dt.getMinute(), dt.getSecond(), dt.getNano() // 1000)
+        else:
+            raise Exception("Unsupported conversion: " + str(type(dt)) + " -> datetime.time")
     except Exception as e:
         raise DHError(e) from e
 
 
-# endregion
-
-# region Conversions: Epoch
-
-def epoch_nanos(dt: Union[Instant, ZonedDateTime]) -> int:
-    """ Returns nanoseconds from the Epoch for a date time value.
-
-    Args:
-        dt (Union[Instant,ZonedDateTime]): Date time.
-
-    Returns:
-        nanoseconds since Epoch, or a NULL_LONG value if the date time is null.
-
-    Raises:
-        DHError
+def to_datetime(dt: Union[None, Instant, ZonedDateTime]) -> Optional[datetime.datetime]:
     """
-    if not dt:
-        return NULL_LONG
-
-    try:
-        return _JDateTimeUtils.epochNanos(dt)
-    except Exception as e:
-        raise DHError(e) from e
-
-
-def epoch_micros(dt: Union[Instant, ZonedDateTime]) -> int:
-    """ Returns microseconds from the Epoch for a date time value.
+    Converts a Deephaven instant or zoned date time to a Python date time.
 
     Args:
-        dt (Union[Instant,ZonedDateTime]): Date time.
+        dt (Union[None, Instant, ZonedDateTime]): A Deephaven instant or zoned date time.  If None is provided, None is
+            returned.
 
     Returns:
-        microseconds since Epoch, or a NULL_LONG value if the date time is null.
-
-    Raises:
-        DHError
+        datetime.datetime
     """
-    if not dt:
-        return NULL_LONG
-
     try:
-        return _JDateTimeUtils.epochMicros(dt)
+        if dt is None:
+            return None
+        elif isinstance(dt, Instant.j_type):
+            ts = dt.getEpochSecond() + (dt.getNano() / 1000000000)
+            return datetime.datetime.fromtimestamp(ts)
+        elif isinstance(dt, ZonedDateTime.j_type):
+            ts = dt.toEpochSecond() + (dt.getNano() / 1000000000)
+            return datetime.datetime.fromtimestamp(ts)
+        else:
+            raise Exception("Unsupported conversion: " + str(type(dt)) + " -> datetime.datetime")
     except Exception as e:
         raise DHError(e) from e
 
 
-def epoch_millis(dt: Union[Instant, ZonedDateTime]) -> int:
-    """ Returns milliseconds from the Epoch for a date time value.
-
-    Args:
-        dt (Union[Instant,ZonedDateTime]): Date time.
-
-    Returns:
-        milliseconds since Epoch, or a NULL_LONG value if the date time is null.
-
-    Raises:
-        DHError
+def to_datetime64(dt: Union[None, Instant, ZonedDateTime]) -> Optional[np.datetime64]:
```

Comment:
`to_np_datetime64`?

Reply:
If there are other datetime64-like things that are also in here. To me `to_datetime64` does not automatically tell me it will be numpy, but maybe I don't know Python well enough.